clippy_lints/src/use_self.rs

```diff
@@ -221,10 +221,10 @@ impl<'a, 'tcx> Visitor<'tcx> for UseSelfVisitor<'a, 'tcx> {
     fn visit_path(&mut self, path: &'tcx Path, _id: HirId) {
         if path.segments.last().expect(SEGMENTS_MSG).ident.name != SelfUpper.name() {
             if self.item_path.def == path.def {
-                span_use_self_lint(self.cx, path);
+                span_use_self_lint(self.cx, path.segments.first().expect(SEGMENTS_MSG).ident.span);
```

Comment:
The first segment might not be the type, but a module instead. So this would replace the module name, instead of the type. [Playground example](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=c7cf1f7ae5b52941ec9423362c4e7bdd)

Reply:
https://github.com/rust-lang/rust-clippy/blob/2e0977f3b4061ed626fe9bfe29c703d13346abc1/clippy_lints/src/use_self.rs#L59-L63

Not sure if putting it in `span_use_self_lint` is the best idea.
If there is any cleaner way to clip `path`, any guidance would be appreciated.
I also added tests to check for this, any better suggestions on naming or more tests would be nice.